src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1492,30 +1492,29 @@ private static LambdaConversionResult IsAnonymousFunctionCompatibleWithDelegate(
                 // SPEC: and modifiers as the corresponding parameter in F.
                 // SPEC: If F has an implicitly typed parameter list, D has no ref or out parameters.
 
+                for (int p = 0; p < delegateParameters.Length; ++p)
```

Comment:
the ref-kind match check is now always performed.  we want to do it for both explicitly and implicitly typed lambdas.